### PR TITLE
store-gateway: remove fine-grained chunks caching

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
#### What this PR does

This is the first PR in a series to remove fine-grained chunks caching - see https://github.com/grafana/mimir/issues/3939#issuecomment-1688474573

In this PR I'm removing 
* the experimental configuration parameter to toggle fine-grained chunks caching `-blocks-storage.bucket-store.chunks-cache.fine-grained-chunks-caching-enabled`
* the experimental configuration parameter to select how many ranges of chunks get cached per series per block `-blocks-storage.bucket-store.fine-grained-chunks-caching-ranges-per-series`
	* because the code is still working with rages of chunks, chunks will now be partitioned into one range per series per block, which was the default value
* the logic to storing and retrieve ranges of chunks from the cache
* related tests

#### Note to reviewers

The two major iterators - `loadingSeriesChunksSetIterator` and `loadingSeriesChunksRefsSetIterator` - have places which can be futher simplified, but I'm leaving that for another PR, so this is easier to review.

Future PRs will also include removing the ranges of chunks abstraction altogether.

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
 * [CHANGE] The `-shutdown-delay` flag is no longer experimental. #5701
 * [CHANGE] The `-validation.create-grace-period` is now enforced in the ingester too, other than distributor and query-frontend. If you've configured `-validation.create-grace-period` then make sure the configuration is applied to ingesters too. #5712
 * [CHANGE] The `-validation.create-grace-period` is now enforced for examplars too in the distributor. If an examplar has timestamp greater than "now + grace_period", then the exemplar will be dropped and the metric `cortex_discarded_exemplars_total{reason="exemplar_too_far_in_future",user="..."}` increased. #5761
+* [CHANGE] Store-gateway: remove fine-grained chunks caching. The following configuration paramters have been removed `-blocks-storage.bucket-store.chunks-cache.fine-grained-chunks-caching-enabled`, `-blocks-storage.bucket-store.fine-grained-chunks-caching-ranges-per-series`.
```
Comment (56quarters) [inline CHANGELOG.md:19]:
```suggestion
* [CHANGE] Store-gateway: remove experimental fine-grained chunks caching. The following experimental configuration parameters have been removed `-blocks-storage.bucket-store.chunks-cache.fine-grained-chunks-caching-enabled`, `-blocks-storage.bucket-store.fine-grained-chunks-caching-ranges-per-series`.
```

**Review by 56quarters — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,7 @@
 * [CHANGE] The `-shutdown-delay` flag is no longer experimental. #5701
 * [CHANGE] The `-validation.create-grace-period` is now enforced in the ingester too, other than distributor and query-frontend. If you've configured `-validation.create-grace-period` then make sure the configuration is applied to ingesters too. #5712
 * [CHANGE] The `-validation.create-grace-period` is now enforced for examplars too in the distributor. If an examplar has timestamp greater than "now + grace_period", then the exemplar will be dropped and the metric `cortex_discarded_exemplars_total{reason="exemplar_too_far_in_future",user="..."}` increased. #5761
+* [CHANGE] Store-gateway: remove experimental fine-grained chunks caching. The following experimental configuration parameters have been removed `-blocks-storage.bucket-store.chunks-cache.fine-grained-chunks-caching-enabled`, `-blocks-storage.bucket-store.fine-grained-chunks-caching-ranges-per-series`.
```
Comment (pracucci) [inline CHANGELOG.md:19]: [nit] Add the PR number, please.

**Review by pracucci — APPROVED** [review]
Thanks for working on this cleanup. LGTM! Looking forward for the next one ;)

File: CHANGELOG.md (hunk 2 above)
Comment (dimitarvdimitrov) [reply CHANGELOG.md:19]:
```suggestion
* [CHANGE] Store-gateway: remove experimental fine-grained chunks caching. The following experimental configuration parameters have been removed `-blocks-storage.bucket-store.chunks-cache.fine-grained-chunks-caching-enabled`, `-blocks-storage.bucket-store.fine-grained-chunks-caching-ranges-per-series`. #5816
```

**Pull request merged**